cmd/bucket-replication.go

```diff
@@ -113,7 +113,7 @@ func validateReplicationDestination(ctx context.Context, bucket string, rCfg *re
 		if clnt == nil {
 			return sameTarget, toAPIError(ctx, BucketRemoteTargetNotFound{Bucket: bucket})
 		}
-		if opts.CheckRemoteBucket { // validate remote bucket
+		if opts != nil && opts.CheckRemoteBucket { // validate remote bucket
```

Comment:
Adding this at the top of the function is simpler and safer:
```go
if opts == nil {
        opts := &validateReplicationDestinationOptions{}
}
```
Or even simpler... Don't make it a pointer, but pass the structure by value.